js/wordsearch.js

```diff
@@ -0,0 +1,346 @@
+      var shiftY = (rowFrom === rowTo) ? 0 : (rowTo > rowFrom) ? 1 : -1,
+        shiftX = (colFrom === colTo) ? 0 : (colTo > colFrom) ? 1 : -1,
+        row = rowFrom,
+        col = colFrom;
+        items.push(this.getItem(row, col));
+        do {
+          row += shiftY;
+          col += shiftX;
+          items.push(this.getItem(row, col));
+        } while( row !== rowTo || col !== colTo );
+    }
+    return items;
+  }
+
+  /**
+   * Returns matrix item
+   * @param {Number} row number
+   * @param {Number} col number
+   * @return {array}
+   */
+  WordSearch.prototype.getItem = function(row, col) {
+    return this.matrix[row] && this.matrix[row][col];
+  };
+
+  // Reset the selected Word 
+  WordSearch.prototype.reset = function() {
+    var selectedWord = document.querySelectorAll('.ws-selected');
+      selectedWord && selectedWord.forEach(word => {
+      word.classList.remove('ws-selected');
+    })
+  }
+
+  /**
+   * Check selected word is present in search wordlist
+   * @param {Array} selected
+   * @param {Array} settings
+   */
+  WordSearch.prototype.searched = function(currentWord,settings) {
+    var words = "";
+    currentWord.forEach(word => {
+        words += word.letter;
+    })
+    var reverseWord=words.split('').reverse().join('');
+    if (this.settings.words.indexOf(words) > -1 ||
+        this.settings.words.indexOf(reverseWord) > -1) {
+        currentWord.forEach(letter => {
+          var current = Object.assign({},letter),
+          wordFound = document.querySelector('.word-search .ws-row:nth-child(' + ++current.row + ') .ws-col:nth-child(' + ++current.col + ')');
+          wordFound.classList.add('ws-found');
+        })
+
+      // Cross word off list.
+      var hintsList = document.querySelector(".hints");
+      var hintsListItems = hintsList.getElementsByTagName("li");
+      for(var hintsListItem = 0; hintsListItem < hintsListItems.length; hintsListItem++){
+        if(words === hintsListItems[hintsListItem].innerHTML.toUpperCase()){
+          if(hintsListItems[hintsListItem].innerHTML != "<del>"+hintsListItems[hintsListItem].innerHTML+"</del>") { // Check the word is never found
+            hintsListItems[hintsListItem].innerHTML = "<del>"+hintsListItems[hintsListItem].innerHTML+"</del>";
+            this.hintFound++;                                       // Increment hintFound.
+          }
+        }
+      }
+
+      // Game over
+      if(this.hintFound === this.settings.words.length){
+        var statusElem = document.createElement("div");
+        statusElem.setAttribute("id", "ws-game-over-outer");
+        statusElem.setAttribute("class", "ws-game-over-outer");
+        this.gridSection.parentNode.appendChild(statusElem);
+        statusElem = document.getElementById("ws-game-over-outer");                // Create statusElem content.
+        var gameOverElem = document.getElementById("game-over-message")           // Show Game Over Message.
```

Comment:
why we are doing this,? move the complete code, create a game over the section in html file, and when the game gets over then just show the game over the section